Blog post: Announcing MyST-JS (#936)
executablebooks/meta: docs/blog/2023/announce-mystjs.md

- **Be community driven**: We believe that standards for scholarly communication require community governance and leadership, rather than being driven by one stakeholder. We framed MyST and the Jupyter Book stack as a collaboration between many organizations and will develop open governance that follows [the principles of open scholarly infrastructure](https://openscholarlyinfrastructure.org/).

The first implementation of MyST was [the MyST Markdown Python parser](https://myst-parser.readthedocs.io), a parser that allows users to take advantage of Sphinx's powerful **directives** and **roles** syntax in building their documentation.

Over the years we have seen interest in MyST raise considerably - it allows authors and researchers to express themselves more efficiently than traditional source text languages like LaTeX.
MyST has also become [a staple in the Python documentation ecosystem](https://www.sphinx-doc.org/en/master/usage/markdown.html), and is the backbone of authoring content in [Jupyter Book](https://jupyterbook.org).

> ## Bringing MyST to JavaScript with a new partnership

Reviewer: 🚀

This effort is based [on their command line interface](https://curvenote.com/docs/cli), which was designed for building technical documentation and computational narratives with JavaScript.
It will become the backbone of **MyST-JS**, and moving forward will be maintained by the Executable Books community.
There is still much to do, but you can preview this work already at [js.myst-tools.org](https://myst-tools.org/docs/mystjs).

Here are a few major new steps summarizing this work over the coming months:

### MyST is now a top-level project in Executable Books

We'll treat MyST as another major effort of the Executable Books project, similar to Jupyter Book.
It will have its own strategy and goals.
Jupyter Book will remain a consumer of MyST markdown, and it will remain one of many stakeholders in the MyST ecosystem.

:::{admonition} A new website!
We have set up a dedicated site to describe the MyST project at [myst-tools.org](https://myst-tools.org).
:::

### MyST will have first-class support in JavaScript

> The [MyST-JS project](https://github.com/executablebooks/mystjs) is written entirelly in TypeScript, with the goal of being re-usable across more applications and interfaces.

Reviewer 2: ```suggestion
The [MyST-JS project](https://github.com/executablebooks/mystjs) is written entirely in TypeScript, with the goal of being re-usable across more applications and interfaces.
```

It also allows us to leverage the vast ecosystem of JavaScript libraries and frameworks for communication.

> For example, we can use [templating frameworks to export MyST documents to a variety of LaTeX publication formats](https://github.com/myst-templates).
> In the future, we hope to leverage the [unifiedjs project](https://unifiedjs.com/) to make MyST documents even more structured and consumable as data.

Reviewer: ```suggestion
For example, we can use [templating frameworks to export MyST documents to a variety of LaTeX publication formats](https://github.com/myst-templates), and leverage the [unifiedjs project](https://unifiedjs.com/) to make MyST documents even more structured and consumable as data.
```

We are already making considerable use of unifiedjs (e.g. all transformations of the AST, parsing latex, serializing tex), we aren't making use of it yet for markdown-parsing which I think is what the future part referred to. I just tweaked the wording slightly as it is all future facing already.

:::{admonition} Learn more
You can find the documentation for MyST JS at [js.myst-tools.org](https://myst-tools.org/docs/mystjs). You can also find an [issue from Rowan discussing major new planned functionality here](https://github.com/executablebooks/meta/issues/838).
:::

> ### MyST will have a Jupyter Lab extension

Reviewer 3: ```suggestion
### MyST will have a JupyterLab extension
```

By bringing MyST into JavaScript, we can integrate it directly with JupyterLab and other web-based Jupyter interfaces.
We have already explored a few ways of doing this via JupyterLab extensions.
The [`jupyterlab-myst` repository](https://github.com/executablebooks/jupyterlab-myst) brings basic functionality of MyST in JupyterLab via `markdown-it` plugins, and the [`jupyterlab-mystjs`](https://github.com/executablebooks/jupyterlab-mystjs) repository is a more fully-functional document engine with MyST in JupyterLab.

> We hope to merge the second repository into the first in the future, so follow [`jupyterlab-myst`](https://github.com/executablebooks/jupyterlab-myst) moving forward.

Reviewer: Depending on when this gets released - @agoose77 and I are having a chat tomorrow to potentially (hopefully!) bring the two efforts together!

Reviewer: This is being merged! See https://github.com/executablebooks/jupyterlab-myst/issues/57 and next comment!

> By bringing MyST into JavaScript, we can integrate it directly with JupyterLab and other web-based Jupyter interfaces.
> We have already explored a few ways of doing this via JupyterLab extensions.
> The [`jupyterlab-myst` repository](https://github.com/executablebooks/jupyterlab-myst) brings basic functionality of MyST in JupyterLab via `markdown-it` plugins, and the [`jupyterlab-mystjs`](https://github.com/executablebooks/jupyterlab-mystjs) repository is a more fully-functional document engine with MyST in JupyterLab.
> We hope to merge the second repository into the first in the future, so follow [`jupyterlab-myst`](https://github.com/executablebooks/jupyterlab-myst) moving forward.

Reviewer: @agoose77 and I have worked today to get these two repositories merged together, we are tracking all outstanding pieces in https://github.com/executablebooks/jupyterlab-myst/issues/57 and will have this fully deployed in the next day or so!

```suggestion
By bringing MyST into JavaScript, we can integrate it directly with JupyterLab and other web-based Jupyter interfaces.
For example, the [`jupyterlab-myst`](https://github.com/executablebooks/jupyterlab-myst) extension brings native MyST rendering into JupyterLab including admonitions, cross-references, and figure numbering using `mystjs` as the document engine.
```

We are still working out the details of how this spec will be structured, and what the process looks like, so please provide feedback if you have ideas for how to build a healthy and inclusive process around the MyST language.

:::{admonition} Learn more
The [MyST Enhancement Proposals repository](https://github.com/executablebooks/myst-enhancement-proposals) has the latest information about the MEP process and any active MEPs.
:::

### MyST-JS will complement Jupyter Book and focus on articles

We think that MyST-JS and Jupyter Book will be able to complement one another in the use-cases they focus on.
The Jupyter Book project (and its Sphinx stack) are focused on building multi-page books and project documentation.
It will continue to lean heavily into the use-cases that [Sphinx](https://sphinx-doc.org) was designed for.
It also has major functionality that is missing in MyST-JS (for example, full internationalization support and extensions).
We will continue to improve and maintain this stack (both as individual projects in the Sphinx ecosystem and as the Jupyter Book distribution).

> The MyST-JS project will initially focus its efforts on a slightly different use-case: **scientific publishing usecases**, such as reproducible analyses meant for publication, and integration with publisher workflows like [JATS](https://en.wikipedia.org/wiki/Journal_Article_Tag_Suite) and [Latex](https://www.latex-project.org/).

Reviewer: ```suggestion
The MyST-JS project will initially focus its efforts on a slightly different use-case: **scientific publishing usecases**, such as reproducible analyses meant for publication, and integration with publisher workflows like [JATS](https://en.wikipedia.org/wiki/Journal_Article_Tag_Suite) and [LaTeX](https://www.latex-project.org/).
```

:)
